static/js/stream_data.js

```diff
@@ -448,17 +450,18 @@ exports.update_calculated_fields = function (sub) {
     sub.can_change_name_description = page_params.is_admin;
     // If stream is public then any user can subscribe. If stream is private then only
     // subscribed users can unsubscribe.
-    // Guest users can't subscribe themselves to any stream.
+    // Guest users can't subscribe themselves to any stream other than web public.
     sub.should_display_subscription_button = sub.subscribed ||
-        !page_params.is_guest && !sub.invite_only;
+        !(page_params.is_guest && !sub.is_web_public) && !sub.invite_only;
     sub.should_display_preview_button = sub.subscribed || !sub.invite_only ||
                                         sub.previously_subscribed;
     sub.can_change_stream_permissions = page_params.is_admin && (
         !sub.invite_only || sub.subscribed);
     // User can add other users to stream if stream is public or user is subscribed to stream.
-    // Guest users can't access subscribers of any(public or private) non-subscribed streams.
-    sub.can_access_subscribers = page_params.is_admin || sub.subscribed || !page_params.is_guest &&
-                                 !sub.invite_only;
+    // Guest users can only access subscribers of web_public streams they can join.
+    sub.can_access_subscribers = page_params.is_admin || sub.should_display_preview_button &&
```

Comment:
This boolean is quickly getting unwieldy, we should break it down.

Reply:
Yes, it was a pain dealing with all those booleans. There were some changes I did that I was afraid of not be doing what I expected, so I ended up doing the truth table just in case.

But after this get done, I was actually going to ask about the possibility of refactoring this and use Enums instead of booleans so it's easier to deal with all those conditions, and you can do changes that are correct.